frang/test_http_resp_code_block.py

```diff
@@ -0,0 +1,62 @@
+"""Functional tests for http_resp_code_block."""
+
+import os
+
+from framework import tester, deproxy_server
+from helpers import tf_cfg, deproxy, tempesta, control
+from framework.templates import fill_template
+
+__author__ = 'Tempesta Technologies, Inc.'
+__copyright__ = 'Copyright (C) 2019 Tempesta Technologies, Inc.'
+__license__ = 'GPL2'
+
+class HttpRespCodeBlock(tester.TempestaTest):
+    backends = [
+        {
+            'id' : 'deproxy',
+            'type' : 'deproxy',
+            'port' : '8000',
+            'response' : 'static',
+            'response_content' : """HTTP/1.1 404 OK
+Content-Length: 0
+
+"""
+        },
+    ]
+
+    tempesta = {
+        'config' : """
+server ${general_ip}:8000;
+
+frang_limits {
+    http_resp_code_block 404 20 5;
+}
+""",
+    }
+
+    clients = [
+        {
+            'id' : 'deproxy',
+            'type' : 'deproxy',
+            'addr' : "${tempesta_ip}",
+            'port' : '80'
+        },
+    ]
+
+    def test(self):
+        requests = ""
+        for i in xrange(1,22):
+            requests += "GET /uri HTTP/1.1\r\n" \
+                       "Host: localhost\r\n" \
+                       "\r\n"
+        deproxy_srv = self.get_server('deproxy')
+        deproxy_srv.start()
+        self.assertEqual(0, len(deproxy_srv.requests))
+        self.start_tempesta()
+        self.assertTrue(deproxy_srv.wait_for_connections(timeout=1))
+        deproxy_cl = self.get_client('deproxy')
+        deproxy_cl.start()
+        deproxy_cl.make_requests(requests)
+        resp = deproxy_cl.wait_for_response(timeout=5)
+        self.assertTrue(resp, "Response not received")
```

Comment:
This is workaround. Need to check that the client connection was really closed. We have no API for that but it extreamly required for all frang tests.